### PR TITLE
Replace C# 10.0 with C# 10.

### PR DESCRIPTION
**Pull request by BillWagner**
Contributes to #26777

### CODE REVIEW

File: docs/csharp/whats-new/tutorials/interpolated-string-handler.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ In this tutorial, you'll learn how to:
 
 ## Prerequisites
 
-You'll need to set up your machine to run .NET 6, including the C# 10.0 compiler. The C# 10 compiler is available starting with [Visual Studio 2022](https://visualstudio.microsoft.com/downloads/) or [.NET Core 6.0 SDK](https://dotnet.microsoft.com/download).
+You'll need to set up your machine to run .NET 6, including the C# 10 compiler. The C# 10 compiler is available starting with [Visual Studio 2022](https://visualstudio.microsoft.com/downloads/) or [.NET Core 6.0 SDK](https://dotnet.microsoft.com/download).
```
Comment (gewarren) [inline docs/csharp/whats-new/tutorials/interpolated-string-handler.md:21]:
```suggestion
You'll need to set up your machine to run .NET 6, including the C# 10 compiler. The C# 10 compiler is available starting with [Visual Studio 2022](https://visualstudio.microsoft.com/downloads/) or [.NET 6 SDK](https://dotnet.microsoft.com/download).
```

File: docs/csharp/whats-new/tutorials/records.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ In this tutorial, you'll learn how to:
 
 ## Prerequisites
 
-You'll need to set up your machine to run .NET 6 or later, including the C# 10.0 or later compiler. The C# 9.0 compiler is available starting with [Visual Studio 2022 preview](https://visualstudio.microsoft.com/vs) or the [.NET 6.0 SDK](https://dotnet.microsoft.com/download).
+You'll need to set up your machine to run .NET 6 or later, including the C# 10 or later compiler. The C# 9.0 compiler is available starting with [Visual Studio 2022 preview](https://visualstudio.microsoft.com/vs) or the [.NET 6.0 SDK](https://dotnet.microsoft.com/download).
```
Comment (gewarren) [inline docs/csharp/whats-new/tutorials/records.md:20]:
```suggestion
You'll need to set up your machine to run .NET 6 or later, including the C# 10 or later compiler. The C# 9.0 compiler is available starting with [Visual Studio 2022 preview](https://visualstudio.microsoft.com/vs) or the [.NET 6 SDK](https://dotnet.microsoft.com/download).
```

File: docs/csharp/whats-new/tutorials/top-level-statements.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ In this tutorial, you'll learn how to:
 
 ## Prerequisites
 
-You'll need to set up your machine to run .NET 6, which includes the C# 10.0 compiler. The C# 10.0 compiler is available starting with [Visual Studio 2022](https://visualstudio.microsoft.com/vs/preview/) or [.NET 6.0 SDK](https://dot.net/get-dotnet6).
+You'll need to set up your machine to run .NET 6, which includes the C# 10 compiler. The C# 10 compiler is available starting with [Visual Studio 2022](https://visualstudio.microsoft.com/vs/preview/) or [.NET 6.0 SDK](https://dot.net/get-dotnet6).
```
Comment (gewarren) [inline docs/csharp/whats-new/tutorials/top-level-statements.md:18]:
```suggestion
You'll need to set up your machine to run .NET 6, which includes the C# 10 compiler. The C# 10 compiler is available starting with [Visual Studio 2022](https://visualstudio.microsoft.com/vs/preview/) or [.NET 6 SDK](https://dot.net/get-dotnet6).
```

File: docs/csharp/language-reference/attributes/general.md [hunk 4]
```diff
@@ -139,11 +139,11 @@ You can learn about async method builders by reading about the following builder
...
+Starting with C# 10, you use these attributes to specify that a type is an *interpolated string handler*. The .NET 6.0 library already includes <xref:System.Runtime.CompilerServices.DefaultInterpolatedStringHandler?displayProperty=nameWithType> for scenarios where you use an interpolated string as the argument for a `string` parameter. You may have other instances where you want to control how interpolated strings are processed. You apply the <xref:System.Runtime.CompilerServices.InterpolatedStringHandlerAttribute?displayProperty=nameWithType> to the type that implements your handler. You apply the <xref:System.Runtime.CompilerServices.InterpolatedStringHandlerArgumentAttribute?displayProperty=nameWithType> to parameters of that type's constructor.
```
Comment (gewarren) [inline docs/csharp/language-reference/attributes/general.md:146]:
```suggestion
Starting with C# 10, you use these attributes to specify that a type is an *interpolated string handler*. The .NET 6 library already includes <xref:System.Runtime.CompilerServices.DefaultInterpolatedStringHandler?displayProperty=nameWithType> for scenarios where you use an interpolated string as the argument for a `string` parameter. You may have other instances where you want to control how interpolated strings are processed. You apply the <xref:System.Runtime.CompilerServices.InterpolatedStringHandlerAttribute?displayProperty=nameWithType> to the type that implements your handler. You apply the <xref:System.Runtime.CompilerServices.InterpolatedStringHandlerArgumentAttribute?displayProperty=nameWithType> to parameters of that type's constructor.
```

File: docs/csharp/language-reference/compiler-options/language.md [hunk 5]
```diff
@@ -105,7 +105,7 @@ The following table lists the minimum versions of the SDK with the C# compiler t
 
 | C# version | Minimum SDK version                                                                  |
 |------------|--------------------------------------------------------------------------------------|
-| C# 10.0    | Microsoft Visual Studio/Build Tools 2022, or .NET 6.0 SDK                            |
+| C# 10    | Microsoft Visual Studio/Build Tools 2022, or .NET 6.0 SDK                            |
```
Comment (gewarren) [inline docs/csharp/language-reference/compiler-options/language.md:108]:
```suggestion
| C# 10     | Microsoft Visual Studio/Build Tools 2022, or .NET 6 SDK                              |
```

File: docs/csharp/language-reference/tokens/interpolated.md [hunk 6]
```diff
@@ -88,7 +88,7 @@ If an interpolated string has the type `string`, it's typically transformed into
...
+Beginning with C# 10, when an interpolated string is used, the compiler checks if the interpolated string is assigned to a type that satisfies the *interpolated string handler pattern*. An *interpolated string handler* is a custom type that converts the interpolated string into a string. An interpolated string handler is an advanced scenario, typically used for performance reasons. You can learn about the requirements to build an interpolated string handler in the language specification for [interpolated string improvements](~/_csharplang/proposals/csharp-10.0/improved-interpolated-strings.md#the-handler-pattern). You can build one following the [interpolated string handler tutorial](../../whats-new/tutorials/interpolated-string-handler.md) in the What's new in C# section. In .NET 6.0, when you use an interpolated string for an argument of type `string`, the interpolated string is processed by the <xref:System.Runtime.CompilerServices.DefaultInterpolatedStringHandler?displayProperty=fullName>.
```
Comment (gewarren) [inline docs/csharp/language-reference/tokens/interpolated.md:91]:
```suggestion
Beginning with C# 10, when an interpolated string is used, the compiler checks if the interpolated string is assigned to a type that satisfies the *interpolated string handler pattern*. An *interpolated string handler* is a custom type that converts the interpolated string into a string. An interpolated string handler is an advanced scenario, typically used for performance reasons. You can learn about the requirements to build an interpolated string handler in the language specification for [interpolated string improvements](~/_csharplang/proposals/csharp-10.0/improved-interpolated-strings.md#the-handler-pattern). You can build one following the [interpolated string handler tutorial](../../whats-new/tutorials/interpolated-string-handler.md) in the What's new in C# section. In .NET 6, when you use an interpolated string for an argument of type `string`, the interpolated string is processed by the <xref:System.Runtime.CompilerServices.DefaultInterpolatedStringHandler?displayProperty=fullName>.
```

File: docs/csharp/whats-new/csharp-10.md [hunk 7]
```diff
@@ -23,7 +23,7 @@ C# 10.0 adds the following features and enhancements to the C# language:
 
 Some of the features you can try are available only when you set your language version to "preview". These features may have more refinements in future previews before .NET 6.0 is released.
```
Comment (gewarren) [inline docs/csharp/whats-new/csharp-10.md:24]:
```suggestion
Some of the features you can try are available only when you set your language version to "preview". These features may have more refinements in future previews before .NET 6 is released.
```

File: docs/csharp/whats-new/csharp-10.md [hunk 8]
```diff
@@ -23,7 +23,7 @@ C# 10.0 adds the following features and enhancements to the C# language:
...
 
-C# 10.0 is supported on **.NET 6**. For more information, see [C# language versioning](../language-reference/configure-language-version.md).
+C# 10 is supported on **.NET 6**. For more information, see [C# language versioning](../language-reference/configure-language-version.md).
 
 You can download the latest .NET 6.0 SDK from the [.NET downloads page](https://dotnet.microsoft.com/download). You can also download [Visual Studio 2022 preview](https://visualstudio.microsoft.com/vs/preview/vs2022/), which includes the .NET 6.0 preview SDK.
```
Comment (gewarren) [inline docs/csharp/whats-new/csharp-10.md:28]:
```suggestion
You can download the latest .NET 6 SDK from the [.NET downloads page](https://dotnet.microsoft.com/download). You can also download [Visual Studio 2022 preview](https://visualstudio.microsoft.com/vs/preview/vs2022/), which includes the .NET 6 preview SDK.
```

File: docs/csharp/whats-new/csharp-10.md [hunk 9]
```diff
@@ -76,14 +76,14 @@ For more information, see the [Extended property patterns](~/_csharplang/proposa
...
+In C# 10, `const` strings may be initialized using [string interpolation](../language-reference/tokens/interpolated.md) if all the placeholders are themselves constant strings. String interpolation can create more readable constant strings as you build constant strings used in your application. The placeholder expressions can't be numeric constants because those constants are converted to strings at runtime. The current culture may affect their string representation. Learn more in the language reference on [`const` expressions](../language-reference/keywords/const.md).
 
 > [!NOTE]
 > When using .NET 6.0 preview 5, this feature requires setting the `<LangVersion>` element in your *csproj* file to `preview`.
```
Comment (gewarren) [inline docs/csharp/whats-new/csharp-10.md:82]: Is this only for preview 5? If so, we should delete it.

File: docs/csharp/whats-new/csharp-10.md [hunk 10]
```diff
@@ -76,14 +76,14 @@ For more information, see the [Extended property patterns](~/_csharplang/proposa
...
 
 ## Record types can seal ToString
 
-In C# 10.0, you can add the `sealed` modifier when you override `ToString` in a record type. Sealing the `ToString` method prevents the compiler from synthesizing a `ToString` method for any derived record types. A `sealed` `ToString` ensures all derived record types use the `ToString` method defined in a common base record type. You can learn more about this feature in the article on [records](../language-reference/builtin-types/record.md).
+In C# 10, you can add the `sealed` modifier when you override `ToString` in a record type. Sealing the `ToString` method prevents the compiler from synthesizing a `ToString` method for any derived record types. A `sealed` `ToString` ensures all derived record types use the `ToString` method defined in a common base record type. You can learn more about this feature in the article on [records](../language-reference/builtin-types/record.md).
 
 > [!NOTE]
 > When using .NET 6.0 preview 5, this feature requires setting the `<LangVersion>` element in your *csproj* file to `preview`.
```
Comment (gewarren) [inline docs/csharp/whats-new/csharp-10.md:89]: Same comment.

**Review by gewarren — APPROVED** [review]

File: docs/csharp/whats-new/csharp-10.md (hunk 9 above)
Comment (BillWagner) [reply docs/csharp/whats-new/csharp-10.md:82]: done.

File: docs/csharp/whats-new/csharp-10.md (hunk 10 above)
Comment (BillWagner) [reply docs/csharp/whats-new/csharp-10.md:89]: done.

**Pull request merged**